pkgs/by-name/ri/ringboard/package.nix

```
@@ -0,0 +1,91 @@
+{
+  rustPlatform,
+  fetchFromGitHub,
+  lib,
+  libxkbcommon,
+  libGL,
+  wayland,
+  xorg,
+  makeWrapper,
+}:
+
+rustPlatform.buildRustPackage (finalAttrs: {
+  pname = "ringboard";
+
+  # release version needs nightly, so we use a custom tree, see:
+  # https://github.com/SUPERCILEX/clipboard-history/issues/22#issuecomment-3322075172
+  version = "0.12.2-unstable-2025-09-23";
+
+  src = fetchFromGitHub {
+    owner = "SUPERCILEX";
+    repo = "clipboard-history";
+    rev = "228a39dd8a9aece0bb06f68ad44906b297270628";
+    hash = "sha256-qA7wwvWnnZHN9edkmubEo37F+peU0LQGo/Zl8FpywuE=";
+  };
+
+  cargoHash = "sha256-MFfuUu/hpb6Uaqe21bvXNKRyJazAL5m+Vw/vAeeDYEk=";
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = [
+    libxkbcommon
+    libGL
+
+    # WINIT_UNIX_BACKEND=wayland
+    wayland
+
+    # WINIT_UNIX_BACKEND=x11
+    xorg.libXcursor
+    xorg.libXrandr
+    xorg.libXi
+    xorg.libX11
+  ];
+
+  buildPhase = ''
+     runHook preBuild
+
+    local flagsArray=("-j $NIX_BUILD_CORES --target x86_64-unknown-linux-gnu --offline --release");
+    concatTo flagsArray cargoBuildFlags;
+
+    echo "Building package: clipboard-history-server"
+    cargo build $flagsArray --package clipboard-history-server --no-default-features --features systemd
+    echo "Building package: clipboard-history-x11"
+    cargo build $flagsArray --package clipboard-history-x11 --no-default-features
+    echo "Building package: clipboard-history-wayland"
+    cargo build $flagsArray --package clipboard-history-wayland --no-default-features
+    echo "Building package: clipboard-history"
+    cargo build $flagsArray --package clipboard-history
+    echo "Building package: clipboard-history-tui"
+    cargo build $flagsArray --package clipboard-history-tui
+    echo "Building package: clipboard-history-egui"
+    cargo build $flagsArray --package clipboard-history-egui
+
+     runHook postBuild
+  '';
+
+  # check needs nightly, see:
+  # https://github.com/SUPERCILEX/clipboard-history/issues/22#issuecomment-3322330559
+  doCheck = false;
+
+  postInstall = ''
+    wrapProgram  $out/bin/ringboard-egui --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath finalAttrs.buildInputs}"
+    install -m 444 -D egui/ringboard-egui.desktop $out/share/applications/ringboard-egui.desktop
+    install -Dm644 logo.jpeg -t $out/share/icons/hicolor/1024x1024/
```

I think it would be better to rename `logo.jpeg` to `ringboard.jpg` when installing so that the icon can be resolved dynamically.

```suggestion
    install -Dm644 logo.jpeg $out/share/icons/hicolor/1024x1024/ringboard.jpg
```